### DISCUSSION TITLE
V2ray 的5.1.0取消了Telegram的﻿MTProto协议支持？

### DISCUSSION THREAD

**Discussion by wind315** [v2fly/v2ray-core#2285] [General]
在新版的配置手册里面的代理协议里面，没有看到如何配置﻿MTProto协议，是取消了Telegram的﻿MTProto协议支持了么？这个协议其实还是很不错的，可以直接通过此协议使用Telegram而不用管其他方面，不知还会加入此协议支持么？

**Comment by hirakujira** [comment]
確實不支援了，config 讀入後出現錯誤

```
Failed to start: main/commands: failed to load config: [/etc/v2ray/config.json] > infra/conf/v4: failed to load inbound detour config. > infra/conf/cfgcommon/loader: unknown config id: mtproto
```

**Comment by kulongwangzhi85** [comment]
sock5不是也可以用吗

**Comment by kingwilliam** [comment]
自建 MTProto Proxy 也不难

https://github.com/9seconds/mtg

**Comment by AkinoKaede** [comment]
Yes, it was removed.

**Comment by Totoro625** [comment]
> 自建 MTProto Proxy 也不难
> 
> https://github.com/9seconds/mtg

试图换到 9seconds/mtg 发现恰好 9seconds/mtg 不能下载媒体及文件了：https://github.com/9seconds/mtg/issues/283
目前切换到 [v4.45.2](https://github.com/v2fly/v2ray-core/releases/tag/v4.45.2) 继续用 MTProto

**Comment by kingwilliam** [comment]
> > 自建 MTProto Proxy 也不难
> > https://github.com/9seconds/mtg
> 
> 试图换到 9seconds/mtg 发现恰好 9seconds/mtg 不能下载媒体及文件了：[9seconds/mtg#283](https://github.com/9seconds/mtg/issues/283) 目前切换到 [v4.45.2](https://github.com/v2fly/v2ray-core/releases/tag/v4.45.2) 继续用 MTProto

试试加入

tcp-buffer = "5mb"
concurrency = 1024

**Comment by maike2019** [comment]
> > 自建 MTProto Proxy 也不难
> > https://github.com/9seconds/mtg
> 
> 试图换到 9seconds/mtg 发现恰好 9seconds/mtg 不能下载媒体及文件了：[9seconds/mtg#283](https://github.com/9seconds/mtg/issues/283) 目前切换到 [v4.45.2](https://github.com/v2fly/v2ray-core/releases/tag/v4.45.2) 继续用 MTProto

请问你是手动搭建的吗？有什么方法能一键自动部署 4.45.2 版本的V2ray吗？

**Comment by hirakujira** [comment]
> > > 自建 MTProto Proxy 也不难
> > > https://github.com/9seconds/mtg
> > 
> > 
> > 试图换到 9seconds/mtg 发现恰好 9seconds/mtg 不能下载媒体及文件了：[9seconds/mtg#283](https://github.com/9seconds/mtg/issues/283) 目前切换到 [v4.45.2](https://github.com/v2fly/v2ray-core/releases/tag/v4.45.2) 继续用 MTProto
> 
> 请问你是手动搭建的吗？有什么方法能一键自动部署 4.45.2 版本的V2ray吗？

docker 指定 image 版本即可
